### ISSUE TITLE
Fails with pragma

### ISSUE THREAD

**Issue by cup** [mklement0/perli#3]
works as expected:

~~~sh
$ perl -e 'use bigint; print 255->to_hex'
ff
~~~

failure:

~~~
perli> use bigint
perli> 255->to_hex
Can't locate object method "to_hex" via package "255" (perhaps you forgot to
load "255"?).
~~~

**Comment by mklement0** [comment]
(I assume you meant `as_hex`.)
The `man` page states (emphasis added):

> **Pragmas such as `use locale;` only take effect for the line at hand**;
	 for pragmas to take effect globally, you must specify them on the
	 command line when starting perli; e.g., `perli -Mlocale`.

Therefore, start `perli` with `perli -Mbigint` to load the bigint module and have it be available session-globally:

 ```sh
$ perli -Mbigint
perli> 255->as_hex
'0xff'
```


Alternatively, for a _single input line_:

 ```sh
# Module bigint is only imported for the line at hand.
perli> use bigint; 255->as_hex
'0xff'
```

**Comment by cup** [comment]
no, i mean `to_hex`:

https://metacpan.org/pod/Math::BigInt#to_hex()

as to the rest of the response, hm, thats not ideal. as python handles it:

~~~py
>>> import datetime
>>> datetime.date.today()
datetime.date(2019, 1, 26)
~~~

and Ruby:

~~~rb
irb(main):001:0> require 'date'
=> true
irb(main):002:0> Date.today
=> #<Date: 2019-01-26 ((2458510j,0s,0n),+0s,2299161j)>
~~~

**Comment by mklement0** [comment]
Yes, it's not ideal - I wish it worked the way you expected and the way Python and Ruby do it, but at least with the current implementation I couldn't make that happen (it's been quite a while since I've looked at it, though).

**Comment by cup** [comment]
It looks like this works as expected with Devel::REPL:

~~~pl
$ re.pl
$ use bigint
$ 255->to_hex
ff
~~~

and Reply:

~~~pl
$ reply
0> use bigint
1> 255->to_hex
$res[0] = 'ff'
~~~

- https://metacpan.org/pod/Devel::REPL
- https://metacpan.org/pod/Reply

**Comment by mklement0** [comment]
Thanks, @cup. I'll reopen this and label it as an enhancement.

I've never looked at these other REPLs (as you can probably tell, I'm not really much of a Perl guy) - have you used them enough to know their relative strengths and how they compare to this project?

**Comment by cup** [comment]
well maybe i am "doing it wrong" - but my primary use case is working from
standard input. for example with python:

~~~py
python3 -i -q <<eof
import datetime
datetime.date.today()
eof
~~~

Result:

~~~py
>>> >>> datetime.date(2019, 1, 26)
>>>
~~~

"Reply" can sort of do this:

~~~py
reply /dev/stdin <<eof
2 + 3;
eof
~~~

but it prints a lot of extra garbage, plus it doesnt exit at EOF, notice
carefully the `1>` in the output:

~~~
0> do "\/dev\/stdin"
$res[0] = 5

1>
~~~

and Devel::REPL cant do standard input at all, and it doesnt exit either:

~~~
$ echo 2+3 | re.pl
$

$ echo 2+3 | re.pl -
$

$ echo 2+3 | re.pl /dev/stdin
$
~~~

**Comment by cup** [comment]
@mklement0 right. `perl` alone would work save for the fact that you need explicit print

probably i will just do that for the time being - but if a proper REPL was made available i would prefer it

**Comment by cup** [comment]
i have another workaround. using `REPL.pl`:

~~~pl
#!/bin/perl
require $ARGV[0];
use feature ':all';
say $px;
~~~

You can use it with any input, as long as the input defines `$px` as some point.
For example:

~~~pl
REPL.pl /dev/stdin <<'eof'
use bigint;
$px = 255->to_hex;
eof
~~~

Result:

~~~
ff
~~~
